### ISSUE TITLE
Errors in buildBench

### ISSUE THREAD

**Issue by patrickrchao** [pkimes/benchmark-fdr#22]
Hello, I am attempting to run some of the dataset rmd files, specifically ```datasets/RNA-Seq/yeast-simulation.Rmd```. Upon reaching the line ```buildBench(data=geneExp, parallel = FALSE)```, I receive the following error:

```
sb <- bd %>% buildBench(data=geneExp, parallel = FALSE)
!! error caught in buildBench !!
!! error in method: bl-df02
!!  original message: 
!!  Error in (~swfdr::lm_pi0)(pValues = ~pval, X = ~ind_covariate, smooth.df = ~2L): unused argument (pValues = ~pval)

!! error caught in buildBench !!
!! error in method: bl-df03
!!  original message: 
!!  Error in (~swfdr::lm_pi0)(pValues = ~pval, X = ~ind_covariate, smooth.df = ~3L): unused argument (pValues = ~pval)

!! error caught in buildBench !!
!! error in method: bl-df04
!!  original message: 
!!  Error in (~swfdr::lm_pi0)(pValues = ~pval, X = ~ind_covariate, smooth.df = ~4L): unused argument (pValues = ~pval)

!! error caught in buildBench !!
!! error in method: bl-df05
!!  original message: 
!!  Error in (~swfdr::lm_pi0)(pValues = ~pval, X = ~ind_covariate, smooth.df = ~5L): unused argument (pValues = ~pval)

Model selection starts!
Shrink the set of candidate models if it is too time-consuming.
  |==================================================| 100%
!! error caught in buildBench !!
!! error in method: fdrreg-t
!!  original message: 
!!  Error in (~FDRreg::FDRreg)(z = ~test_statistic, features = ~model.matrix(~splines::bs(ind_covariate, : unused arguments (features = ~model.matrix(~splines::bs(ind_covariate, df = 3) - 1), control = ~list(lambda = 0.01))

!! error caught in buildBench !!
!! error in method: fdrreg-e
!!  original message: 
!!  Error in (~FDRreg::FDRreg)(z = ~test_statistic, features = ~model.matrix(~splines::bs(ind_covariate, : unused arguments (features = ~model.matrix(~splines::bs(ind_covariate, df = 3) - 1), control = ~list(lambda = 0.01))
```

Thank you,
Patrick

**Comment by pkimes** [comment]
Can you share the output of `sessionInfo()`?

I'm guessing this is because you're using a newer version of the `swfdr` package than what was out when we ran the benchmark. We ran the experiments with the following version of the package (it's recorded in the `benchmarkfdrData2019` package):

```
> library("benchmarkfdrData2019")
> dat <- `baxter-genus-benchmark`()
> colData(dat)["bl-df02", c("pkg_name", "pkg_vers")]

DataFrame with 1 row and 2 columns
           pkg_name    pkg_vers
        <character> <character>
bl-df02       swfdr       1.6.0
```

It looks like the newest version of the `swfdr` package replaced the `pValues=` argument with `p=` for better readability (unfortunate that it's backwards incompatible...).

- [Bioc 3.9 package docs](https://bioconductor.org/packages/3.9/bioc/manuals/swfdr/man/swfdr.pdf)
- [Bioc 3.10 package docs](https://bioconductor.org/packages/3.10/bioc/manuals/swfdr/man/swfdr.pdf)

You should be able to fix this by changing one line in the `initializeBenchDesign` function in the `common-BenchDesign.R` where the benchmark design is constructed and the `swfdr::lm_pi0()` parameters are specified. [Change this line](https://github.com/pkimes/benchmark-fdr/blob/18c1e412afbbbbfa742a3b4c124c9e1611d5a4e4/datasets/R/common-BenchDesign.R#L86) from:
```
                       pValues = pval, X = ind_covariate,
```
to:
```
                       p = pval, X = ind_covariate,
```
Basically, map the values in the `pval` column of your data (`geneExp`) to the `p=` parameter rather than the (nw non-existant) `pValues=` parameter.

For the `FDRreg` issues, try installing the repo from the author's GitHub repo (https://github.com/jgscott/FDRreg).
```
remotes::install_github('jgscott/FDRreg', subdir="R_pkg/")
```
The package on CRAN is outdated.

**Comment by patrickrchao** [comment]
Thanks for the insight!

Here is my session info:
```
sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.5

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] splines   parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] SummarizedBenchmark_0.99.2  adaptMT_1.0.0               FDRreg_0.1                  fda_5.1.4                   Matrix_1.2-18              
 [6] fdrtool_1.2.15              swfdr_1.12.0                qvalue_2.18.0               ashr_2.2-47                 IHW_1.14.0                 
[11] R.utils_2.9.2               R.oo_1.23.0                 R.methodsS3_1.8.0           ggthemes_4.2.0              DESeq2_1.26.0              
[16] purrr_0.3.4                 cowplot_1.0.0               magrittr_1.5                readr_1.3.1                 data.table_1.12.8          
[21] tibble_3.0.1                crayon_1.3.4                sessioninfo_1.1.1           digest_0.6.25               dplyr_0.8.5                
[26] mclust_5.4.6                ggplot2_3.3.0               stringr_1.4.0               rlang_0.4.5                 UpSetR_1.4.0               
[31] SummarizedExperiment_1.16.1 DelayedArray_0.12.3         BiocParallel_1.20.1         matrixStats_0.56.0          Biobase_2.46.0             
[36] GenomicRanges_1.38.0        GenomeInfoDb_1.22.1         IRanges_2.20.2              S4Vectors_0.24.4            BiocGenerics_0.32.0        
[41] tidyr_1.0.2                

loaded via a namespace (and not attached):
  [1] backports_1.1.6        Hmisc_4.4-0            plyr_1.8.6             lazyeval_0.2.2         crosstalk_1.1.0.1      leaflet_2.0.3         
  [7] usethis_1.6.0          lpsymphony_1.14.0      invgamma_1.1           htmltools_0.4.0        SQUAREM_2020.2         fansi_0.4.1           
 [13] checkmate_2.0.0        memoise_1.1.0          mosaicData_0.17.0      cluster_2.1.0          remotes_2.1.1          mosaicCore_0.6.0      
 [19] annotate_1.64.0        prettyunits_1.1.1      jpeg_0.1-8.1           colorspace_1.4-1       blob_1.2.1             ggrepel_0.8.2         
 [25] xfun_0.13              callr_3.4.3            RCurl_1.98-1.2         genefilter_1.68.0      survival_3.1-12        glue_1.4.0            
 [31] polyclip_1.10-0        gtable_0.3.0           zlibbioc_1.32.0        XVector_0.26.0         ggformula_0.9.4        pkgbuild_1.0.6        
 [37] scales_1.1.0           DBI_1.1.0              mosaic_1.6.0           Rcpp_1.0.4.6           xtable_1.8-4           htmlTable_1.13.3      
 [43] ggstance_0.3.4         foreign_0.8-76         bit_1.1-15.2           Formula_1.2-3          truncnorm_1.0-8        htmlwidgets_1.5.1     
 [49] RColorBrewer_1.1-2     acepack_1.4.1          ellipsis_0.3.0         pkgconfig_2.0.3        XML_3.99-0.3           farver_2.0.3          
 [55] nnet_7.3-13            locfit_1.5-9.4         tidyselect_1.0.0       labeling_0.3           reshape2_1.4.4         AnnotationDbi_1.48.0  
 [61] munsell_0.5.0          tools_3.6.3            cli_2.0.2              generics_0.0.2         RSQLite_2.2.0          devtools_2.3.0        
 [67] broom_0.5.6            ggdendro_0.1-20        fs_1.4.1               processx_3.4.2         knitr_1.28             bit64_0.9-7           
 [73] nlme_3.1-147           slam_0.1-47            compiler_3.6.3         rstudioapi_0.11        curl_4.3               png_0.1-7             
 [79] testthat_2.3.2         tweenr_1.0.1           geneplotter_1.64.0     stringi_1.4.6          ps_1.3.2               desc_1.2.0            
 [85] lattice_0.20-41        vctrs_0.2.4            pillar_1.4.3           lifecycle_0.2.0        BiocManager_1.30.10    bitops_1.0-6          
 [91] irlba_2.3.3            R6_2.4.1               latticeExtra_0.6-29    gridExtra_2.3          MASS_7.3-51.5          assertthat_0.2.1      
 [97] pkgload_1.0.2          rprojroot_1.3-2        withr_2.2.0            GenomeInfoDbData_1.2.2 hms_0.5.3              grid_3.6.3            
[103] rpart_4.1-15           mixsqp_0.3-17          ggforce_0.3.1          base64enc_0.1-3 
```
Upon changing the arguments, the ```swfdr``` seems to be working now! However I am having a great deal of trouble installing ```FDRreg```. After messing around with the path, the following command does allow for the installation to begin, however there is an error with installation.

```
remotes::install_github('jgscott/FDRreg', subdir="/R_pkg")
Downloading GitHub repo jgscott/FDRreg@master
✓  checking for file ‘/private/var/folders/p2/qjns7zj118l97lkbqykpkg7c0000gn/T/RtmpsiXa0t/remotes110042fca0a88/jgscott-FDRreg-9197f4b/R_pkg/DESCRIPTION’ ...
─  preparing ‘FDRreg’:
✓  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘FDRreg_0.2-1.tar.gz’
   
* installing *source* package ‘FDRreg’ ...
** using staged installation
** libs
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/ -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RcppArmadillo/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c FDRreg.cpp -o FDRreg.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/ -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RcppArmadillo/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c RNG.cpp -o RNG.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/ -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RcppArmadillo/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c RRNG.cpp -o RRNG.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/ -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RcppArmadillo/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c RcppExports.cpp -o RcppExports.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/ -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RcppArmadillo/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c fl_dp.cpp -o fl_dp.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/ -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RcppArmadillo/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c normix.cpp -o normix.o
clang++ -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o FDRreg.so FDRreg.o RNG.o RRNG.o RcppExports.o fl_dp.o normix.o -L/Library/Frameworks/R.framework/Resources/lib -lRlapack -L/Library/Frameworks/R.framework/Resources/lib -lRblas -L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin15/6.1.0 -L/usr/local/gfortran/lib -lgfortran -lquadmath -lm -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
ld: warning: directory not found for option '-L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin15/6.1.0'
ld: warning: directory not found for option '-L/usr/local/gfortran/lib'
ld: library not found for -lgfortran
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [FDRreg.so] Error 1
ERROR: compilation failed for package ‘FDRreg’
* removing ‘/Library/Frameworks/R.framework/Versions/3.6/Resources/library/FDRreg’
Error: Failed to install 'FDRreg' from GitHub:
  (converted from warning) installation of package ‘/var/folders/p2/qjns7zj118l97lkbqykpkg7c0000gn/T//RtmpsiXa0t/file110046caa6891/FDRreg_0.2-1.tar.gz’ had non-zero exit status
```

It seems likely that this is an issue on the ```FDRreg``` end.

**Comment by pkimes** [comment]
To compile the package, you'll need a fortran compiler (e.g. `gfortran`). A few ways you can solve this. Try taking a look at these SO answers.

- https://stackoverflow.com/questions/35999874
- https://stackoverflow.com/questions/29992066

Basically, you need to install `gfortran` with homebrew or something else and change the R package compiler parameters in `~/.R/Makevars` (explained in the SO answers ^) to know where to find `gfortran`. Hope this helps.

**Comment by patrickrchao** [comment]
That fixed things, thank you!

I have begun running the rest of the file and it seems that most things work well. 

The main thing I have encountered is related to the ```summarizedBenchmark``` package, in the```tidyUpMetrics``` function. I believe the ```gather``` function now takes in ```key``` instead of ```keys``` and the default column name has also changed.

Thank you again!

**Comment by pkimes** [comment]
Great!

Yeah - we fixed the issue related to the `tidyr` update in the release branch of the `SummarizedBenchmark` package, but not in the branch for the benchmarking project. Is this just a warning or an error? Sounds like you were able to fix it.

**Comment by patrickrchao** [comment]
This was an error that I was unable to fix. However it seemed to be solely related to ```rejections_scatter```(although I have not tested all of the other methods since they require far more computation) so I have skipped running ```rejections_scatter``` for now. Is there a branch for ```benchmark-fdr``` updated for ```tidyr```?

**Comment by pkimes** [comment]
There isn't an update for this in the [`benchmarkfdr` branch](https://github.com/areyesq89/SummarizedBenchmark/tree/fdrbenchmark) of the `SummarizedBenchmark` package, but I can try to update this over the weekend since (I think) it should be pretty straightforward. Should be fine if you just copy over [the `tidyUpMetrics` function from the newest version of the package](https://github.com/areyesq89/SummarizedBenchmark/blob/071123dbdd/R/PerformanceMetrics.R#L310-L333). 

Can you open an issue over on the `SummarizedBenchmark` repo?

**Comment by patrickrchao** [comment]
Yes, copying the ```tidyUpMetrics``` function works! Thank you!

**Comment by pkimes** [comment]
awesome! thanks for trying.
will try to get around to updating this in the `benchmarkfdr` branch at some point.
